### PR TITLE
Breakout temperature alerts to entity

### PR DESCRIPTION
**Pull request by disforw**
This PR will now create a binary sensor entity for each flood/temperature sensor that alerts on determined threshold. Set your threshold in the Bhyve app.

### CODE REVIEW

**Comment by disforw** [comment]
Do not merge until review requested, still testing…

**Comment by disforw** [comment]
All tests complete

File: custom_components/bhyve/binary_sensor.py [hunk 1]
```diff
@@ -89,3 +90,41 @@ def _on_ws_data(self, data):
 
     def _should_handle_event(self, event_name, data):
         return event_name in [EVENT_FS_ALARM]
+
+    
+class BHyveTemperatureAlert(BHyveDeviceEntity):
```
Comment (sebr) [inline custom_components/bhyve/binary_sensor.py:95]: This should be renamed:

```suggestion
class BHyveTemperatureBinarySensor(BHyveDeviceEntity):
```

File: custom_components/bhyve/binary_sensor.py [hunk 2]
```diff
@@ -89,3 +90,41 @@ def _on_ws_data(self, data):
...
+        name = "{} temperature alert".format(device.get("name"))
+        super().__init__(hass, bhyve, device, name, "alert")
+
+    def _setup(self, device):
+        self._available = device.get("is_connected", False)
+        self._state = self._parse_status(device.get("status", {}))
+    
+    def _parse_status(self, status):
+        """Convert BHyve alarm status to entity value."""
+        return "on" if status.get("temp_alarm_status") == "high_temp_alarm" else "off"
```
Comment (sebr) [inline custom_components/bhyve/binary_sensor.py:107]: As far as I can tell, the device also allows setting a low temperature alert.

I'm unsure the right way to model this in Home Assistant. I can see one of two options:
1. One `TemperatureBinarySensor` where `state() -> "on"` if `temp < low || temp > high`. Entity attributes would have the appropriate attributes for `temp_threshold_low`, `temp_threshold_low_exceeded`, `temp_threshold_high` and `temp_threshold_high_exceeded`
2. Two binary sensors: `LowTemperatureBinarySensor` and `HighTemperatureBinarySensor` (should be self explanatory)

File: custom_components/bhyve/binary_sensor.py [hunk 3]
```diff
@@ -89,3 +90,41 @@ def _on_ws_data(self, data):
...
+    
+class BHyveTemperatureAlert(BHyveDeviceEntity):
+
+    def __init__(self, hass, bhyve, device):
+        name = "{} temperature alert".format(device.get("name"))
+        super().__init__(hass, bhyve, device, name, "alert")
+
+    def _setup(self, device):
+        self._available = device.get("is_connected", False)
+        self._state = self._parse_status(device.get("status", {}))
```
Comment (sebr) [inline custom_components/bhyve/binary_sensor.py:103]: Should add the attributes for the threshold temperatures and any other relevant attributes

**Review by sebr — CHANGES_REQUESTED** [review]
Have left a few comments.

Have you discovered any API endpoints which can be called to change the thresholds? These would be useful to include as `service` endpoints. You can poke around at https://techsupport.orbitbhyve.com/ with the web inspector to try and uncover.

As well, if you could please update the README as relevant that would be appreciated.

File: custom_components/bhyve/binary_sensor.py (hunk 2 above)
Comment (disforw) [reply custom_components/bhyve/binary_sensor.py]: I’m going to go with your first suggestion. The idea being that this entity is in alarm. This is how bHyve handles this in their app.

Comment (disforw) [reply custom_components/bhyve/binary_sensor.py]: This should do it…
return "on" if "alarm" in status.get("temp_alarm_status") else "off"

File: custom_components/bhyve/binary_sensor.py (hunk 3 above)
Comment (disforw) [reply custom_components/bhyve/binary_sensor.py:103]: That’s a great idea, only I dont think they provide this in the API. I dont see it.

Comment (disforw) [reply custom_components/bhyve/binary_sensor.py:103]: SCRATCH THAT!! I found it, attributes added as requested.

**Comment by disforw** [comment]
Made all requested changes,I looked in to changing the limits but can't find anything. If I find something in the future about it, I'll submit it as another pr.

**Review by disforw — COMMENTED** [review]
Ready to go

File: custom_components/bhyve/binary_sensor.py (hunk 3 above)
Comment (sebr) [reply custom_components/bhyve/binary_sensor.py:103]: Thanks! Can you please upload your diagnostics for the device so I can look at the properties?

**Comment by disforw** [comment]
{
        "last_connected_at": "2022-04-19T10:54:48.368Z",
        "location_name": "Laundry Room",
        "firmware_version": "16",
        "temp_alarm_thresholds": {
            "high_temp_f": 85,
            "low_temp_f": 55
        },
        "name": "Washing Machine",
        "type": "flood_sensor",
        "battery": {
            "percent": 100
        },
        "updated_at": "2023-05-28T17:42:58.221Z",
        "reference": "xxxxxxxxxx",
        "mac_address": "xxxxxxxx",
        "status": {
            "last_flood_alarm_at": "2023-05-20T01:49:21.951Z",
            "rssi": -84,
            "temp_f": 71.6,
            "provisioned": true,
            "phy": "le_1m_1000",
            "temp_alarm_status": "ok",
            "status_updated_at": "2023-05-28T17:42:58.220Z",
            "flood_alarm_status": "ok"
        },
        "id": "xxxxxxxxxxx",
        "auto_shutoff": false,
        "user_id": "xxxxxxx",
        "device_gateway_topic": "devices-2",
        "hardware_version": "fs1-0001",
        "is_connected": true,
        "created_at": "2021-08-26T21:47:11.293Z",
        "network_topology_id": "xxxxxxxxxxxxx"
    },

**Review by sebr — APPROVED** [review]

**Pull request merged**